### ISSUE TITLE
is this virus definition Img.Packed.PngContainsDownloadCmd-6786216-0 reintroduced in the database

### ISSUE THREAD

**Issue by akshitjain03** [Cisco-Talos/clamav#1283]
Describe the bug
----------------
Started 19th June 2024, we started seeing cases where the clamAV started flagging png files as infected using this virus definition:  **Img.Packed.PngContainsDownloadCmd-6786216-0**
Our system fetches the latest clamav database everyday. Since June 19th we started seeing this issue.

**[Update]**
I unpacked the daily.cvd files for june18/june19/june20
In unpacked daily.cvd received from public servers on june19 , i see the daily.ign2 file contains only this string "fake_dont_remove_this_line" while daily.ign2 for june18 and june20 are correct.

**Does this means clamav released a borken/corrupted daily.cvd on june19?**

PFA all three daily.cvd (unpacked) for june18,june19 and june20
https://drive.google.com/drive/folders/1wrxIm_WRc8Wo3UCH-rg37_KrVan_5nJ1?usp=sharing 

How to reproduce the problem
----------------------------
Download the daily.cvd from test19 folder shared in the google drive link
Try some png files with ClamAV and latest database , one should be able to see it flagged with Img.Packed.PngContainsDownloadCmd-6786216-0 virus definition.

Please check the clamAV DB and see if this definition Img.Packed.PngContainsDownloadCmd-6786216-0 has been introduced again or its a mistake?

**Comment by micahsnyder** [comment]
There was an issue publishing the morning of June 19th. We manually published later that day and it seems there was a mistake made in that manual process. Specifically, the `daily.ign2` signature file was missing from `daily.cvd`. This file ignores signatures in the `main.cvd` which have yet to be dropped (we don't update main.cvd very often).

It shouldn't happen again. Sorry for the inconvenience.

**Comment by akshitjain** [comment]
looks like this virus definition Img.Packed.PngContainsDownloadCmd-6786216-0 is reintroduced today? can you please check and remove it

**Comment by val-ms** [comment]
@akshitjain thank you for saying something. I see the issue as well. I will ask for help.

**Comment by val-ms** [comment]
@akshitjain Looking closer at it, there is some inconsistency in our build system internal tagging for `Img.Packed.PngContainsDownloadCmd-6786216-0`.  However, it will not be present if you have the latest daily.cvd and the latest main.cvd.

`Img.Packed.PngContainsDownloadCmd-6786216-1` was added to replace `Img.Packed.PngContainsDownloadCmd-6786216-0` in 2022. An ignore-signature was added at that time to prevent the older `Img.Packed.PngContainsDownloadCmd-6786216-0` revision, which still existed in main.cvd, from being loaded.

Yesterday, we built a new main.cvd, removing `Img.Packed.PngContainsDownloadCmd-6786216-0` entirely. Today, we published a new daily.cvd that removed the ignore-signature.

If you're scanning with the older main.cvd and the newest daily.cvd, you will see `Img.Packed.PngContainsDownloadCmd-6786216-0`

TL;DR: Please run freshclam (or cvdupdate).  In case you have been using some other system to update databases, be sure to use those tools to keep your daily, main, and bytecode signatures up to do date.

**Comment by akshitjain** [comment]
@val-ms i have taken the latest version of main.cvd and daily.cvd on 17th December. After that only the issue started appearling.
Is it possible that latest published main.cvd still contains `Img.Packed.PngContainsDownloadCmd-6786216-0 `
Pls recheck once.

**Comment by val-ms** [comment]
@akshitjain I have double-checked. That signature no longer exists in daily.cvd or main.cvd.  This is what i ran to check it:
```sh
~
❯ cvd update
2025-12-18 09:59:18 AM - INFO:  Using system configured nameservers
2025-12-18 09:59:18 AM - INFO:  main.cvd is up-to-date. Version: 63
2025-12-18 09:59:18 AM - INFO:  Downloaded daily-27854.cdiff
2025-12-18 09:59:19 AM - INFO:  Downloaded daily-27854.cdiff.sign. Version: 27854
2025-12-18 09:59:21 AM - INFO:  Downloaded daily.cvd. Version: 27854
2025-12-18 09:59:21 AM - INFO:  Downloaded daily-27854.cvd.sign. Version: 27854
2025-12-18 09:59:21 AM - INFO:  bytecode.cvd is up-to-date. Version: 339

~ took 4s
❯ mkdir main-unpacked-2

~
❯ cd main-unpacked-2/

~/main-unpacked-2
❯ ~/clams/1.5.1/bin/sigtool --unpack ~/.cvdupdate/database/main.cvd

~/main-unpacked-2
❯ ~/clams/1.5.1/bin/sigtool --unpack ~/.cvdupdate/database/daily.cvd

~/main-unpacked-2
❯ ls
COPYING     daily.fp    daily.hsb   daily.ign2  daily.mdb   daily.ndb   daily.wdb   main.hdb    main.mdb
daily.cdb   daily.ftm   daily.hsu   daily.info  daily.mdu   daily.ndu   main.cdb    main.hsb    main.msb
daily.cfg   daily.hdb   daily.idb   daily.ldb   daily.msb   daily.pdb   main.crb    main.info   main.ndb
daily.crb   daily.hdu   daily.ign   daily.ldu   daily.msu   daily.sfp   main.fp     main.ldb    main.sfp

~/main-unpacked-2
❯ grep -r "Img.Packed.PngContainsDownloadCmd-6786216-0" .

~/main-unpacked-2
❯
```

If I remove the "`-0`" in my search, it finds the newer revision of the signature:
```sh
~/main-unpacked-2
❯ grep -r "Img.Packed.PngContainsDownloadCmd-6786216" .
./daily.ldb:Img.Packed.PngContainsDownloadCmd-6786216-1;Engine:81-255,Target:5;0&(1|2);0:89504E47;6375726c20{1-12}68747470;7767657420{1-12}68747470
```
